src/Effects/RAVU/ravu-r2.hlsl

```diff
@@ -0,0 +1,510 @@
+// Generated by ravu.py
+// 
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+// 
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+// 
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+
+
+
```

Comment:
There are too many blank lines. Please arrange the line breaks and indents reasonably to make the code easier to read.

Reply:
I have fixed places where unnecessary line breaks were generated. I have also added some line breaks in other places to improve readability.

Architecture of the scripts which generate these shaders is not really conducive to generating readable code so the only feasible way to improve readability further seems to be post-processing the code through something like clang-format. I have implemented support for running it as a post-processing step.
Would current formatting be fine or should I tweak .clang-format somehow?

Reply:
In my opinion, the modified codes are already very readable, and I’m not sure if clang-format supports formatting hlsl files.

Reply:
clang-format has no problem formatting .hlsl files.
The code in the last commit has gone through clang-format with the following config: [.clang-format](https://github.com/hauuau/magpie-prescalers/blob/source-magpie-v4/.clang-format)

Reply:
Great!